src/types/markers.js

```diff
@@ -724,6 +724,16 @@ export type MediaSampleMarkerPayload = {|
  */
 export type JankPayload = {| type: 'Jank' |};
 
+export type BrowsertimeMarkerPayload = {|
+  type: 'VisualMetricProgress',
+  percentage: number,
+|};
+
+export type NoPayloadUserData = {|
+  type: 'NoPayloadUserData',
+  innerWindowID?: number,
+|};
```

Comment:
I'm somewhat puzzled by this type. Is that necessary for the tests? or something else?

Reply:
It's the payload type of `Navigation::Start`. I added it to be able to add this marker to our tests. We could use another type for the test too but wanted it to represent a real marker payload. See it here: https://searchfox.org/mozilla-central/rev/fadd0a14d2a2724ee4733ef73970a2ddd457a43f/mozglue/baseprofiler/public/BaseProfilerMarkersDetail.h#229-243

Reply:
ah I see, I didn't know this was "the real thing", good then!

Reply:
Yeah, backend automatically puts this payload when the marker doens't have one but still needs to put the additional data.